lambdapi.opam

```diff
@@ -20,7 +20,7 @@ license: "CeCILL 2.1"
 doc: "https://deducteam.github.io/lambdapi/"
 
 depends: [
-  "ocaml"        {         >= "4.04.0" }
```

Comment:
Why?

Reply:
@gabrielhdt , should this conversation be closed?

Reply:
closed if the above is closed isn'it?

Reply:
I am not sure that it is related. Here, we say that LP needs OCaml 4.04.0 at least. Did you try whether it compiles with older versions?

Reply:
Not sure that it is useful to check this. So, we could keep this constraint.

Reply:
It won't compile as long as I use `Seq` or `choose_opt` stuff

Reply:
Sure. But this is opam's job to take care of computing dependencies. This constraint here means that we didn't check whether LP compiles with older version of OCaml. But I don't mind to remove it. @rlepigre , what do you think?

Reply:
This line goes with the other discussion on the CI file, they should be changed together, whatever the decision is. It is good practice to have version bounds everywhere.

Reply:
Ok, so let's keep this constraint. @gabrielhdt , thanks to fix your PR.